data_safe_haven/pulumi/components/sre_gitea_server.py

```diff
@@ -353,11 +353,13 @@ def __init__(
             ],
             opts=ResourceOptions.merge(
                 ResourceOptions(
+                    delete_before_replace=True,
                     depends_on=[
                         file_share_gitea_caddy_caddyfile,
                         file_share_gitea_gitea_configure_sh,
                         file_share_gitea_gitea_entrypoint_sh,
-                    ]
+                    ],
+                    replace_on_changes=["containers"],
```

Comment:
This marks that resources must be replaced when another is updated/changed?

Is "containers" a resource name or a tag?

Reply:
It means that the `ContainerInstance` resource needs to be replaced whenever there is a change to the `containers` attribute. Without this, `pulumi` will try to do an update-in-place which isn't supported for this resource.